> More work to go on tests it looks like. Do you know how to run them locally?

The "stable" failed at "Download docker images" stage. That's nothing I can control via the bats file.
```
Error response from daemon: unauthorized: authentication required
Failed to RunSimpleContainer to inspect database version/type: failed to pull image busybox:stable: exit status 1, output=
Error: Process completed with exit code 1.
```

Can you run it again?

The "HEAD" is working with 3/4 successful tests. The last one is failing as announced:
```
cp: cannot stat '.ddev/elasticsearch/docker-compose.elasticsearch8.yaml': No such file or directory
Error: Process completed with exit code 1.
```
That will only work after merge/release.

--- a/README.md
+++ b/README.md
@@ -6,6 +6,10 @@ Uses [elasticsearch official image](https://hub.docker.com/_/elasticsearch)
 
 `ddev get ddev/ddev-elasticsearch`
 
+### Using Elasticsearch 8
+
+Post `ddev get`, run `cp .ddev/elasticsearch/docker-compose.elasticsearch8.yaml .ddev/` to enable Elasticsearch 8.
+
 ## Configuration
 
 From within the container, the elasticsearch container is reached at hostname: "elasticsearch", port: 9200, so the server URL might be `http://elasticsearch:9200`. You can also use the "ddev.site" http and https urls to access it: `http://<projectname>.ddev.site:9200`, and `https://<projectname>.ddev.site:9201`

--- a/docker-compose.elasticsearch.yaml
+++ b/docker-compose.elasticsearch.yaml
@@ -22,7 +22,7 @@ services:
       - elasticsearch:/usr/share/elasticsearch/data
       - ".:/mnt/ddev_config"
     healthcheck:
-      test: ["CMD-SHELL", "curl --fail -s localhost:9200"]
+      test: ["CMD-SHELL", "curl --fail -s elasticsearch:9200"]
 
 volumes:
   elasticsearch:

--- a/elasticsearch/config/elasticsearch8.yml
+++ b/elasticsearch/config/elasticsearch8.yml
@@ -1,0 +1,15 @@
+#ddev-generated
+# This file contains the configuration settings for Elasticsearch 8.
+# For more information, see https://www.elastic.co/guide/en/elasticsearch/reference/current/settings.html
+
+# https://www.elastic.co/guide/en/elasticsearch/reference/current/important-settings.html#cluster-name
+cluster.name: "docker-cluster"
+
+# https://www.elastic.co/guide/en/elasticsearch/reference/current/important-settings.html#network.host
+network.host: 0.0.0.0
+
+# Disable security features
+# https://www.elastic.co/guide/en/elasticsearch/reference/current/security-settings.html#general-security-settings
+xpack.security.enabled: false
+xpack.security.autoconfiguration.enabled: false
+xpack.security.enrollment.enabled: false

--- a/elasticsearch/docker-compose.elasticsearch8.yaml
+++ b/elasticsearch/docker-compose.elasticsearch8.yaml
@@ -1,0 +1,6 @@
+#ddev-generated
+services:
+  elasticsearch:
+    image: elasticsearch:8.10.2
+    volumes:
+      - ./elasticsearch/config/elasticsearch8.yml:/usr/share/elasticsearch/config/elasticsearch.yml

--- a/install.yaml
+++ b/install.yaml
@@ -4,4 +4,5 @@ pre_install_actions:
 
 # list of files and directories listed that are copied into project .ddev directory
 project_files:
+- elasticsearch/
 - docker-compose.elasticsearch.yaml

--- a/tests/test.bats
+++ b/tests/test.bats
@@ -1,38 +1,54 @@
 setup() {
   set -eu -o pipefail
   export DIR="$( cd "$( dirname "$BATS_TEST_FILENAME" )" >/dev/null 2>&1 && pwd )/.."
-  export TESTDIR=~/tmp/testelasticsearch
+  export TESTDIR=~/tmp/test-elasticsearch
   mkdir -p $TESTDIR
-  export PROJNAME=testelasticsearch
+  export PROJNAME=test-elasticsearch
+  export ADDON_PATH="ddev/ddev-elasticsearch"
+  export USE_VERSION8=false
   export DDEV_NON_INTERACTIVE=true
-  ddev delete -Oy ${PROJNAME} || true
-  cd "${TESTDIR}"
-  ddev config global --simple-formatting
-  ddev config --project-name=${PROJNAME} --project-type=php
-  ddev start -y
+  ddev delete -Oy ${PROJNAME} >/dev/null 2>&1 || true
+  cd ${TESTDIR} || ( printf "unable to cd to ${TESTDIR}\n" && exit 1 )
+  ddev config --project-name=${PROJNAME}
+  ddev start -y >/dev/null
+}
+
+execute_test() {
+  echo "# ddev get ${ADDON_PATH} with project ${PROJNAME} in ${TESTDIR} ($(pwd))" >&3
+  ddev get ${ADDON_PATH} >/dev/null
+  if [ "${USE_VERSION8}" = true ]; then
+    cp .ddev/elasticsearch/docker-compose.elasticsearch8.yaml .ddev/
+  fi
+  ddev restart >/dev/null
+  health_checks
+}
+
+health_checks() {
+  ddev exec "curl -s elasticsearch:9200" | grep "${PROJNAME}-elasticsearch"
 }
 
 teardown() {
-  set -eu -o pipefail
-  cd ${TESTDIR} || ( printf "unable to cd to ${TESTDIR}\n" && exit 1 )
-  ddev delete -Oy ${PROJNAME}
+  ddev delete -Oy ${PROJNAME} >/dev/null 2>&1
   [ "${TESTDIR}" != "" ] && rm -rf ${TESTDIR}
 }
 
 @test "install from directory" {
-  set -eu -o pipefail
-  cd ${TESTDIR} || ( printf "unable to cd to ${TESTDIR}\n" && exit 1 )
-  echo "# ddev get ${DIR} with project ${PROJNAME} in ${TESTDIR} ($(pwd))" >&3
-  ddev get ${DIR}
-  ddev restart
-  ddev exec "curl -s elasticsearch:9200" | grep "${PROJNAME}-elasticsearch"
+  ADDON_PATH="${DIR}"
+  execute_test
 }
 
 @test "install from release" {
-  set -eu -o pipefail
-  cd ${TESTDIR} || ( printf "unable to cd to ${TESTDIR}\n" && exit 1 )
-  echo "# ddev get ddev/ddev-elasticsearch with project ${PROJNAME} in ${TESTDIR} ($(pwd))" >&3
-  ddev get ddev/ddev-elasticsearch
-  ddev restart
-  ddev exec "curl -s elasticsearch:9200" | grep "${PROJNAME}-elasticsearch"
+  execute_test
 }
+
+@test "install version 8 from directory" {
+  ADDON_PATH="${DIR}"
+  USE_VERSION8=true
+  execute_test
+}
+
+# enable after release of https://github.com/ddev/ddev-elasticsearch/pull/16
+# @test "install version 8 from release" {
+#   USE_VERSION8=true
+#   execute_test
+# }

